### DISCUSSION TITLE
Clear output from long running task

### DISCUSSION THREAD

**Discussion by kayleg** [stevearc/overseer.nvim#190] [Q&A]
Is it possible to clear the output from a long running task without needing to restart the task? Like the equivalent of <Cmd-K> in a standard mac terminal.

I leave a hot-reload server process running and sometimes I log some debug info but it gets really hard to find my latest output and I have to restart the task just to clear the output.

**Comment by stevearc** [comment]
This is _possible_ with the `jobstart` strategy. I'm not sure how you want to trigger it, but you can pipe the raw terminal escape codes to the terminal handle. If you have the task object:
```lua
vim.api.nvim_chan_send(task.strategy.term_id, "\027[H\027[2J")
```

**Comment by kayle-at-built** [comment]
Thanks! This worked great.

Incase other people find this discussion, I implemented the trigger as an action for the task list and made all auto detected tasks use the `jobstart` strategy:


```
      require("overseer").setup({
        actions = {
          ["clear output"] = {
            desc = "Clear the output",
            run = function(task)
              vim.api.nvim_chan_send(task.strategy.term_id, "\027[H\027[2J")
            end,
          },
        },
        task_list = {
          bindings = {
            ['<C-c>'] = '<CMD>OverseerQuickAction clear output<CR>',
          }
        }
      })
      require('overseer').add_template_hook({}, function(task_defn)
        task_defn.strategy = { "jobstart", preserver_output = false, use_terminal = true }
      end
      )
```
